perf: server side render settings/organizations/general

retrogtx opened the pull request:
## What does this PR do?
 - Moved data fetching logic from client-side hooks (`useQuery`, `useSession`) in the `LegacyPage` component (`packages/.../general.tsx`) to the server component `Page` (`apps/web/.../general/page.tsx`).
- Implemented server-side data fetching in `Page.tsx` using the tRPC caller (`appRouter.createCaller`) to retrieve user and organization details.
- Passed the fetched data (`currentOrg`, `isAdminOrOwner`, `localeProp`) down as props to the `LegacyPage` component.
Removed redundant client-side data fetching hooks and related logic from the `LegacyPage` component.



https://github.com/user-attachments/assets/53646eb0-84bc-4561-96c3-43edade69777


    
<!-- This is an auto-generated description by mrge. -->
---

## Summary by mrge
Moved organization general settings data fetching from the client to the server for faster page loads and better security.

- **Refactors**
  - Server component now fetches user and organization data and passes it as props.
  - Removed client-side data fetching and related hooks from the settings page.

<!-- End of auto-generated description by mrge. -->

retrogtx committed:
fix: trigger lingo.dev by removing duplicate value

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
Merge branch 'calcom:main' into main

retrogtx committed:
feat: Enhance organization settings page with server-side session handling

- Implement server-side session validation and redirection for unauthorized users.
- Fetch current organization and user locale from the server.
- Pass organization and user data to the LegacyPage component for improved rendering.
- Refactor OrgGeneralView to accept props for current organization and user permissions.

retrogtx commented on apps/web/app/(use-page-wrapper)/settings/(settings-layout)/organizations/general/page.tsx:
```diff
@@ -13,11 +21,29 @@
+  const ctx = await createContextInner({ session, locale: session.user.locale ?? "en" });
+  const caller = appRouter.createCaller(ctx);
+
+  const user = await caller.viewer.me.get();
```
damn you're right

retrogtx committed:
refactor: Optimize data fetching in organization settings page

hbjORbj commented on apps/web/app/(use-page-wrapper)/settings/(settings-layout)/organizations/general/page.tsx:
```diff
@@ -13,11 +21,31 @@ export const generateMetadata = async () =>
+    redirect("/auth/login");
+  }
+
+  const ctx = await createContextInner({ session, locale: session.user.locale ?? "en" });
+  const caller = appRouter.createCaller(ctx);
```
I created a util called `createRouterCaller` :) You can pass organizations router to it and use it. This way, we don't need to load all routers but only the router you need.

e.g., `  const caller = await createRouterCaller(eventTypesRouter);`

retrogtx replied:
ah I see

the example is right from `apps/web/app/(use-page-wrapper)/(main-nav)/event-types/page.tsx`

doing it

retrogtx committed:
refactor: use util createRouterCaller as benny suggested

retrogtx committed:
add skeleton and loading state

retrogtx committed:
refactor: Consolidate organization settings page structure and remove unused components

- Simplified the loading component import path.
- Merged functionality from the deleted main-page into the general page for better organization.
- Enhanced the page with server-side session handling and data fetching.
- Removed obsolete skeleton and main-page files to streamline the codebase.

retrogtx committed:
refactor: Remove Suspense loading wrapper from organization settings page and header

- Eliminated the Suspense component from the organization settings page and SettingsHeader to simplify rendering.
- Streamlined the code by directly rendering child components without fallback loading states.

hbjORbj commented on apps/web/app/(use-page-wrapper)/settings/(settings-layout)/organizations/general/page.tsx:
```diff
@@ -13,13 +21,26 @@ export const generateMetadata = async () =>
+  const [user, currentOrg] = await Promise.all([meCaller.get(), orgCaller.listCurrent()]);
+
+  if (!currentOrg) {
+    redirect("/getting-started");
```
should be /enterprise according to the client component?

retrogtx replied:
Ah true

I confused this with `!session` in my head probably, that's why redirected to `/getting-started`

Low sleep issues 🫠

retrogtx committed:
fix: Update redirection path when no organisation is found

retrogtx committed:
add back Suspense loading wrapper in SettingsHeader

hbjORbj reviewed (changes requested):
A few pointers:

1. I see that you have a skeleton UI for license key container as well. Is this needed? To my understanding, if license isn't there, user gets routed to /enterprise, anyways. So it is better to not have a skeleton UI for license key container to prevent a layout shift. Let me know if I am missing something here.

2. We are fetching the entire user object to get the locale only.
Rather, let's use:
```
  const locale = (await getLocale(buildLegacyRequest(await headers(), await cookies()))) ?? "en";
```

3. You need to create `actions.ts` and use revalidatePath to invalidate orgs data in this RSC whenever needed.

e.g., Search for how `revalidateEventTypeEditPage` is used throughout the codebase

anikdhabal commented:
Closing due to inactivity. Feel free to tag me if you wanted to continue 🙏
